File: python/프로그래머스/level.1/행렬의 덧셈.py
arr1, arr2 = [[1, 2], [2, 3]], [[3, 4], [5, 6]]
# arr1, arr2 = [[1], [2]], [[3], [4]]

# sol.2
for i in range(len(arr1)):
    for j in range(len(arr1[0])):
        arr1[i][j] += arr2[i][j]
print(arr1)

# sol.1 내풀이
# ans = []
# for i in range((len(arr1))):
#     tmp = []
#     for j in range((len(arr1[0]))):
#         tmp.append(arr1[i][j] + arr2[i][j])
#     ans.append(tmp)
# print(ans)

File: JS/프로그래머스/행렬의 덧셈.js
function solution(arr1, arr2) {
    var answer = [];
    for (let i = 0; i < arr1.length; i++) {
        let temp = [];
        for (let j = 0; j < arr1[i].length; j++) {
            temp.push(arr1[i][j] + arr2[i][j])
        }
        answer.push(temp)
    }
    return answer;
}

function solution(arr1, arr2) {
    return arr1.map((a, i) => a.map((b, j) => b + arr2[i][j]));
}